src/pkg/services/m365/api/channels.go

```diff
@@ -1 +1,376 @@
 package api
+
+import (
+	"context"
+
+	"github.com/alcionai/clues"
+	"github.com/alcionai/corso/src/internal/common/ptr"
+	"github.com/alcionai/corso/src/internal/m365/graph"
+	"github.com/alcionai/corso/src/pkg/backup/details"
+	"github.com/alcionai/corso/src/pkg/fault"
+	"github.com/alcionai/corso/src/pkg/logger"
+	"github.com/microsoft/kiota-abstractions-go/serialization"
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+	"github.com/microsoftgraph/msgraph-sdk-go/teams"
+)
+
+// ---------------------------------------------------------------------------
+// controller
+// ---------------------------------------------------------------------------
+
+func (c Client) Channels() Channels {
+	return Channels{c}
+}
+
+// Channels is an interface-compliant provider of the client.
+type Channels struct {
+	Client
+}
+
+// ---------------------------------------------------------------------------
+// containers
+// ---------------------------------------------------------------------------
+
+// CreateContainer makes an channels with the name in the team
+func (c Channels) CreateChannel(
+	ctx context.Context,
+	teamID, _, containerName string,
+) (graph.Container, error) {
+	body := models.NewChannel()
+	body.SetDisplayName(&containerName)
+
+	container, err := c.Stable.
+		Client().
+		Teams().
+		ByTeamId(teamID).
+		Channels().
+		Post(ctx, body, nil)
+	if err != nil {
+		return nil, graph.Wrap(ctx, err, "creating channel")
+	}
+
+	return ChannelsDisplayable{Channelable: container}, nil
+}
+
+// DeleteChannel removes a channel from user's M365 account
+func (c Channels) DeleteChannel(
```

Comment:
Similarly, do we expect to delete channels?  I can't think of any reason we'd want to do this, whether as part of backup or restore.  We might be better off without this.